modules/c/nrt/include/nrt/Utils.h

```diff
@@ -173,5 +173,24 @@ NRTPROT(void) nrt_Utils_decimalLonToGeoCharArray(double decimal, char *buffer8);
  */
 NRTAPI(char) nrt_Utils_cornersTypeAsCoordRep(nrt_CornersType type);
 
+
+/*!
+ * Helper function to actually perform a byte-swap.
+ *
+ * \param value Pointer to value being swapped
+ * \param indexOne Index of first byte to be swapped
+ * \param indexTwo Index of second byte to be swapped
+ */
+
+NRTPROT(void) nrt_Utils_swap(nrt_Uint8 *value, size_t indexOne, size_t indexTwo);
```

Comment:
I like it.  But... we need to make sure the compiler inlines this, and more than likely it won't when the implementation is in a c file.  So, put the implementation of this in the header and put the `inline` keyword in there to be sure.